shippers/es.py

```diff
@@ -243,26 +295,56 @@ def flush(self) -> None:
         return
 
     def _send_dead_letter_index(self, actions: list[Any]) -> list[Any]:
+        """
+        Send the failed actions to the dead letter index (DLI).
+
+        This function attempts to forward failed actions to the DLI, but may not do so
+        for one of the following reasons:
+
+        1. The action response does not have an HTTP status (e.g., the connection failed).
+        2. The list of action errors to forward is not empty, and the action error type is not in the list.
+        3. The action could not be encoded for indexing in the DLI.
+        4. The action failed indexing attempt in the DLI.
+
+        Args:
+            actions (list[Any]): A list of actions to be processed.
+
+        Returns:
+            list[Any]: A list of actions that were not indexed in the DLI.
```

Comment:
Why do we want to return the list of actions that were not indexed? I can't find its use in the code, but I might be missing it.

Reply:
The `_send_dead_letter_index()` aims to index errors in the DLI.

However, it may not be able to index errors for different reasons:

1. the function fails to encode the error
2. ES returns an error on the DLI indexing request
3. the error is retryable

So, the function returns the list of errors to the caller, saying, "These are the errors I could not (or won't) index on the DLI."

In this case, the caller (`flush()` function ) will send these errors to the replay queue.

The function structure didn't change. I added reason 3 ("the error is retryable") and opted for a more generic `non_indexed_actions` variable name instead of `dead_letter_errors` due to the change in semantics (we don't index errors only due to problems but also as a deliberate decision based on retriability).

Reply:
> and opted for a more generic non_indexed_actions variable name instead of dead_letter_errors

I see :sweat_smile: I thought now these errors were meant to be ignored, hence why I got confused